libs/entitlement/qa/entitlement-test-plugin/src/main/java/org/elasticsearch/entitlement/qa/test/URLConnectionNetworkActions.java

```diff
@@ -79,6 +79,17 @@ private static void withJdkHttpConnection(CheckedConsumer<HttpURLConnection, Exc
         }
     }
 
+    private static void withJdkMailToConnection(CheckedConsumer<URLConnection, Exception> connectionConsumer) throws Exception {
+        var conn = EntitledActions.createMailToURLConnection();
+        // Be sure we got the connection implementation we want
+        assert conn.getClass().getSimpleName().equals("MailToURLConnection");
+        try {
+            connectionConsumer.accept(conn);
+        } catch (IOException e) {
+            // It's OK, it means we passed entitlement checks, and we tried to perform some IO
```

Comment:
In other cases, I've put a `return` in the `catch` block and a `fail()` afterward, to insist that it ought to get an `IOException`. Not sure that matters though: it's not the job of this code to ensure that `accept` works correctly.

Reply:
I'll keep it this way for now if you don't mind, for uniformity with the rest of the methods.
But I made a note to do a pass and review all IT tests after GA: we need to finish moving them out the rest handler, give them uniform names, and in general give them a pass. Better to do everything in one go (or multiple, but close in time and purpose)